packages/nitro-rpc-client/src/serde.ts

```diff
@@ -0,0 +1,260 @@
+import Ajv, { JTDDataType } from "ajv/dist/jtd";
+
+import {
+  ChannelStatus,
+  LedgerChannelInfo,
+  PaymentChannelInfo,
+  RPCRequestAndResponses,
+  RequestMethod,
+} from "./types";
+
+const ajv = new Ajv();
+
+const jsonRpcSchema = {
+  properties: {
+    jsonrpc: { type: "string" },
+    id: { type: "uint32" },
+    result: {
+      properties: {},
+      additionalProperties: true,
+    },
+  },
+  optionalProperties: { error: { type: "string", nullable: true } },
```

Comment:
Should the `error` property be of type `object` (reference: [jsonrpc 2.0 spec](https://www.jsonrpc.org/specification))? Consider changing `error` to something like this (not sure if the nested optional/required properties are conflicting in this case):
```
optionalProperties: { 
  error: { 
    type: "object",
    nullable: true,
    properties: {
      code: { type: "int" },
      message: { type: "string" },
    },
    additionalProperties: true,
    required: ["code", "message"]
  }
},

```

Reply:
Good point! The error schema is updated to adhere to the jsonrpc spec in 31310dd.